scrapy/spider.py

```diff
@@ -31,11 +31,19 @@ def __init__(self, name=None, **kwargs):
         if not hasattr(self, 'start_urls'):
             self.start_urls = []
 
-    def log(self, message, level=log.DEBUG, **kw):
-        """Log the given messages at the given log level. Always use this
-        method to send log messages from your spider
+    @property
+    def logger(self):
+        logger = logging.getLogger(self.name)
```

Comment:
this will break logging for existing spiders, for example a lot of user that are using dash currently will experience unexpected disappearance of spider logs (including quite important WARNING and ERROR level logs) and to fix that they all would need to setup log handlers. Don't you think that by default it would be better to provide logger that would be handled by scrapy and only if user refuses to use default logger and creates new one (which is advanced usage which will be available only after merging this PR) - then he will take all responsibility for creating appropriate handlers?

It's also not clear for me how dash itself would handle spider logs after this change. @krotkiewicz / @dangra are you ready for this?